### ISSUE TITLE
Attempting to build android samples fails with IllegalArgumentException

### ISSUE THREAD

**Issue by ekuefler** [bazelbuild/bazel#405]
I'm unable to build the android sample following the instructions on the site, despite using a fresh bazel installation and adding the android rules to my workspace. My bash history looks like this:

```
git clone https://github.com/google/bazel
cd bazel/
./compile.sh
emacs WORKSPACE 
bazel build //examples/android/java/bazel:hello_world
```

This gives me the following output:

```
$ bazel build //examples/android/java/bazel:hello_world
.....
Unhandled exception thrown during build; message: Unrecoverable error while evaluating node 'CONFIGURED_TARGET:@android_tools//src/tools/android/java/com/google/devtools/build/android/incrementaldeployment:stub_application 38406ebb102d458bcf22edd87324eb5e' (requested by nodes 'CONFIGURED_TARGET:@android_tools//tools/android:incremental_stub_application 38406ebb102d458bcf22edd87324eb5e')
INFO: Elapsed time: 2.329s
java.lang.RuntimeException: Unrecoverable error while evaluating node 'CONFIGURED_TARGET:@android_tools//src/tools/android/java/com/google/devtools/build/android/incrementaldeployment:stub_application 38406ebb102d458bcf22edd87324eb5e' (requested by nodes 'CONFIGURED_TARGET:@android_tools//tools/android:incremental_stub_application 38406ebb102d458bcf22edd87324eb5e')
	at com.google.devtools.build.skyframe.ParallelEvaluator$Evaluate.run(ParallelEvaluator.java:762)
	at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$2.run(AbstractQueueVisitor.java:436)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IllegalArgumentException: android_library does not have access to class com.google.devtools.build.lib.rules.java.JavaConfiguration
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:145)
	at com.google.devtools.build.lib.analysis.RuleContext.getFragment(RuleContext.java:263)
	at com.google.devtools.build.lib.rules.java.JavaHelper.filterLauncherForTarget(JavaHelper.java:69)
	at com.google.devtools.build.lib.rules.java.JavaHelper.launcherForTarget(JavaHelper.java:40)
	at com.google.devtools.build.lib.rules.java.JavaCommon.collectTargetsTreatedAsDeps(JavaCommon.java:516)
	at com.google.devtools.build.lib.rules.java.JavaCommon.<init>(JavaCommon.java:114)
	at com.google.devtools.build.lib.rules.android.AndroidLibrary.create(AndroidLibrary.java:158)
	at com.google.devtools.build.lib.rules.android.AndroidLibrary.create(AndroidLibrary.java:54)
	at com.google.devtools.build.lib.analysis.ConfiguredTargetFactory.createRule(ConfiguredTargetFactory.java:249)
	at com.google.devtools.build.lib.analysis.ConfiguredTargetFactory.createConfiguredTarget(ConfiguredTargetFactory.java:167)
	at com.google.devtools.build.lib.skyframe.SkyframeBuildView.createConfiguredTarget(SkyframeBuildView.java:428)
	at com.google.devtools.build.lib.skyframe.ConfiguredTargetFunction.createConfiguredTarget(ConfiguredTargetFunction.java:711)
	at com.google.devtools.build.lib.skyframe.ConfiguredTargetFunction.compute(ConfiguredTargetFunction.java:186)
	at com.google.devtools.build.skyframe.ParallelEvaluator$Evaluate.run(ParallelEvaluator.java:726)
	... 4 more
```

**Comment by lberki** [comment]
Yep, I found this just a few hours ago and thought I could push a quick fix, but it didn't quite fit in my workday. Expect a fix to be pushed tomorrow morning in Europe.

Now that we have a continuous build, we can actually have some integration tests which will help us avoid such embarrassments in the future.

**Comment by ulfjack** [comment]
I sent out a fix.

**Comment by lberki** [comment]
Ulf seems to have fixed this in 4ac3c22a4.
